### ISSUE TITLE
[feature request] Add repository id field in files data frame

### ISSUE THREAD

**Issue by zurk** [src-d/engine#172]
If you run this code

```
frc = engine.repositories.filter("is_fork = false").references.head_ref.commits.first_reference_commit
frc.printSchema()
frc.files.printSchema()
```
you get this output:
```
root
 |-- repository_id: string (nullable = false)
 |-- reference_name: string (nullable = false)
 |-- index: integer (nullable = false)
 |-- hash: string (nullable = false)
 |-- message: string (nullable = false)
 |-- parents: array (nullable = true)
 |    |-- element: string (containsNull = false)
 |-- tree: map (nullable = true)
 |    |-- key: string
 |    |-- value: string (valueContainsNull = false)
 |-- blobs: array (nullable = true)
 |    |-- element: string (containsNull = false)
 |-- parents_count: integer (nullable = false)
 |-- author_email: string (nullable = true)
 |-- author_name: string (nullable = true)
 |-- author_date: timestamp (nullable = true)
 |-- committer_email: string (nullable = true)
 |-- committer_name: string (nullable = true)
 |-- committer_date: timestamp (nullable = true)

root
 |-- file_hash: string (nullable = false)
 |-- content: binary (nullable = true)
 |-- commit_hash: string (nullable = false)
 |-- is_binary: boolean (nullable = false)
 |-- path: string (nullable = true)
```

I wonder why `files` do not have `repository_id`. It good to know from which repository I get the file.
I think it is a good idea to include this field to `FilesDataFrame`.

**Comment by zurk** [comment]
For now I find this workaround 
```python
frc = engine.repositories.filter("is_fork = false").references.head_ref.commits.first_reference_commit
frc.files.join(frc.select("repository_id", col("hash").alias("commit_hash")), "commit_hash").show()
```

**Comment by erizocosmico** [comment]
We're gonna change a bit the tables and split the files table into `tree_entries` and `blobs`. Since there is no way to guarantee uniqueness of blobs and tree entries without doing a distinct (and thus throwing to the trash our GitOptimizer) we're going to propagate commit hash, repository id and reference name all the way to the blob.

TL;DR: yes, this will be included in the PR i'm currently working on.

**Comment by erizocosmico** [comment]
This is addressed in #200 and will be available (amongst some other breaking changes) when it's merged.
